### PR TITLE
feature(content): add post, read and delete content

### PR DESCRIPTION
**Pull request by Mofuko1316**
## Description
Roughly complete the feature of posting, reading and deleting subject and reply content.
- [x] content post: receive corresponding data of subject post and save in database
- [x] content reply: receive corresponding data of reply of a specific post and save in database
- [x] content read: receive post id and page information of a specific subject post, return corresponding data of subject post with no more than 10 replies in one page
- [x] subject post delete: receive post id and delete subject post as well as its replies in database
- [x] reply delete: receive post id and reply id and delete reply of a specific subject post in database

## How Has This Been Tested
Send request with corresponding data using APIfox.
1. request for posting content (with corresponding data/ data with missing parts)
2. request for replying a specific post (with corresponding data/ data with missing parts)
3. request for reading a specific post with no more than 10 replies in one page
4. request for deleting a post (with post id that exists/ not exists)
5. request for deleting a reply of a specific post (with post id and reply id that exists/ not exists)

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [x] Code follows the code style of this project.
- [x] Changes follow the **CONTRIBUTING** guidelines.
- [x] Update necessary documentation accordingly.
- [x] Lint and tests pass locally with the changes.
- [x] Check issues and pull requests first. You don't want to duplicate effort.

## Other information

### CODE REVIEW

File: backend/src/routes/content.rs [hunk 1]
```diff
@@ -0,0 +1,413 @@
...
+            .expect("Cannot save content");
+        // println!("{:?}", res2.reply_id.unwrap());
+        // return the response
+        post_response.success = true;
+        post_response.post_id = res2.post_id.unwrap();
+        (Status::Ok, Json(post_response))
+    }
+}
+
+#[get("/<post_id>/<page>")]
```
Comment (BobAnkh) [inline backend/src/routes/content.rs:135]: change uri to "`/post/<post_id>?<page>`", use a query string format

File: backend/src/routes/content.rs [hunk 2]
```diff
@@ -0,0 +1,413 @@
...
+        // }else{
+        //     subject_info.author = String::new(subject_info.author);
+        // }
+
+        // return the response
+        (Status::Ok, Json(read_response))
+    }
+}
+
+#[post("/<post_id>/reply", data = "<reply_info>", format = "json")]
```
Comment (BobAnkh) [inline backend/src/routes/content.rs:254]: change uri to "/post/<post_id>/reply"

File: backend/src/routes/content.rs [hunk 3]
```diff
@@ -0,0 +1,413 @@
...
+                .expect("Cannot update content");
+            println!("{:?}", res2.post_id.unwrap());
+            reply_response.success = true;
+            // return the response
+            (Status::Ok, Json(reply_response))
+        }
+    }
+}
+
+#[delete("/<post_id>/delete")]
```
Comment (BobAnkh) [inline backend/src/routes/content.rs:332]: change uri to "/post/<post_id>"

File: backend/src/routes/content.rs [hunk 4]
```diff
@@ -0,0 +1,413 @@
...
+            .expect("cannot delete content from content_reply");
+        // return the response
+        post_response.success = true;
+        post_response.post_id = post_id;
+        (Status::Ok, Json(post_response))
+    }
+}
+
+#[delete("/<post_id>/<reply_id>/delete")]
+pub async fn content_delete_reply(
```
Comment (BobAnkh) [inline backend/src/routes/content.rs:375]: change uri to "/post/<post_id>/reply/<reply_id>", information can be carried in json data body

**Review by BobAnkh — CHANGES_REQUESTED** [review]
Roughly review all the interface, not the code

Please polish the interfaces and **do a cargo clippy check** locally to improve code. Also can see what github actions provide you, or auto comments.

File: backend/src/req/content.rs [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    pub title: String,
+    pub author: String,
+    pub anonymous: bool,
+    pub created_time: String,
+    pub modified_time: String,
+    pub section: String,
+    pub tag1: String,
+    pub tag2: String,
+    pub tag3: String,
+    pub post_len: i32,
```
Comment (Mofuko1316) [inline backend/src/req/content.rs:57]: remove anonymous, merge tag1~tag3

File: backend/src/routes/content.rs [hunk 6]
```diff
@@ -0,0 +1,413 @@
...
+        if subject_data.post_len < page * item_display {
+            reply_info = ContentReply::find()
+                .filter(
+                    Condition::all()
+                        .add(pgdb::content_reply::Column::PostId.eq(post_id))
+                        .add(pgdb::content_reply::Column::ReplyId.gte((page - 1) * item_display)),
+                )
+                .order_by_asc(pgdb::content_reply::Column::ReplyId)
+                .all(&pg_con)
+                .await
```
Comment (BobAnkh) [inline backend/src/routes/content.rs:184]: You can use paginator:

https://www.sea-ql.org/SeaORM/docs/basic-crud/select#paginate-result

https://docs.rs/sea-orm/0.3.2/sea_orm/struct.Paginator.html

**Comment by BobAnkh** [comment]
# TODO

## Valid burrow interface

## Cargo test consumer

**Comment by BobAnkh** [comment]
# TODO

## Check consumer in relation write

**Pull request closed without merging**